homeassistant/components/wireguard/sensor.py

```diff
@@ -0,0 +1,169 @@
+"""Support for WireGuard binary sensors."""
+from datetime import datetime
+
+from ha_wireguard_api.model import WireGuardPeer
+
+from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfInformation
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import WireGuardUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the WireGuard binary sensors based on a config entry."""
+    coordinator: WireGuardUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    sensors: list[SensorEntity] = []
+    sensors.extend(
+        WireGuardPeerHandshakeSensor(coordinator, peer) for peer in coordinator.data
+    )
+    sensors.extend(
+        WireGuardPeerBytesReceivedSensor(coordinator, peer) for peer in coordinator.data
+    )
+    sensors.extend(
+        WireGuardPeerBytesSentSensor(coordinator, peer) for peer in coordinator.data
+    )
+    async_add_entities(sensors)
+
+
+class WireGuardPeerHandshakeSensor(
+    CoordinatorEntity[WireGuardUpdateCoordinator], SensorEntity
+):
+    """Representation of a WireGuard latest_handshake status."""
+
+    _attr_device_class = SensorDeviceClass.TIMESTAMP
+    _attr_has_entity_name = True
+
+    def __init__(self, coordinator: WireGuardUpdateCoordinator, peer_id: str) -> None:
+        """Initialize the WireGuard latest_handshake Sensor."""
+        super().__init__(coordinator)
+        self._peer_id: str = peer_id
+        self._attr_name = "Latest Handshake"
```

Comment:
Please move the names to `strings.json`, so that they can be translated.

Reply:
so.. replace `self._attr_name = "Received"` with `self._name_translation_key = "received"` and add `received` to `strings.json`, correct?